packages/ui/src/views/run-pipeline/utils/inputs-utils.ts

```diff
@@ -4,19 +4,95 @@ import * as z from 'zod'
 
 import { IInputDefinition, unsetEmptyStringOutputTransformer } from '@harnessio/forms'
 
+import { type InputLayout } from './types'
+
 /** pipeline inputs to form inputs conversion */
-export function pipelineInputs2FormInputs(
-  pipelineInputs: Record<string, any>,
+export function pipelineInputs2FormInputs({
+  pipelineInputs,
+  options,
+  pipelineInputLayout = []
+}: {
+  pipelineInputs: Record<string, any>
   options: { prefix?: string }
-): IInputDefinition[] {
-  const formInputs: IInputDefinition[] = []
+  pipelineInputLayout?: InputLayout
+}): IInputDefinition[] {
+  /**
+   * Pre-process inputs for valid layout.
+   * Rule 1 - If there are duplicate keys in the layout, we skip the layout and return all inputs flat
```

Comment:
we should not skip the entire layout, we should only skip the repeated inputs (show them only once)

Reply:
Done, added it as a rule

```
* Rule 1 - If input keys are duplicated in the layout, only the first occurrence is considered
```